src/Analyzer.cpp

```diff
@@ -0,0 +1,369 @@
+
+    // allocate buffer for output from the FFT object
+    mAnalysisBuffers = new float*[fftChans];
+    for (int i = 0; i < fftChans; i++) {
+        mAnalysisBuffers[i] = new float[mAnalysisBuffersSize];
+    }
+
+    // allocate buffers for holding on to past spectra
+    mSpectra              = new float*[mNumSpectra];
+    mSpectraDifferentials = new float*[mNumSpectra];
+    for (int i = 0; i < mNumSpectra; i++) {
+        mSpectra[i]              = new float[fftChans];
+        mSpectraDifferentials[i] = new float[fftChans];
+    }
+}
+
+//*******************************************************************************
+Analyzer::~Analyzer()
+{
+    int fftChans = static_cast<fftdsp*>(mFftP)->getNumOutputs();
+    for (int i = 0; i < fftChans; i++) {
+        delete mAnalysisBuffers[i];
+    }
+
+    for (int i = 0; i < mNumSpectra; i++) {
+        delete mSpectra[i];
+        delete mSpectraDifferentials[i];
+    }
+
+    delete mAnalysisBuffers;
+    delete mSpectra;
+    delete mSpectraDifferentials;
+    delete static_cast<fftdsp*>(mFftP);
+}
+
+//*******************************************************************************
+void Analyzer::init(int samplingRate)
+{
+    ProcessPlugin::init(samplingRate);
+    if (samplingRate != fSamplingFreq) {
+        std::cerr << "Sampling rate not set by superclass!\n";
+        std::exit(1);
+    }
+    fs = float(fSamplingFreq);
+
+    static_cast<fftdsp*>(mFftP)->init(fs);
+
+    /* Start timer */
+    connect(&mTimer, &QTimer::timeout, this, &Analyzer::onTick);
+    mTimer.setTimerType(Qt::PreciseTimer);
+    mTimer.setInterval(mInterval);
+    mTimer.setSingleShot(false);
+    mTimer.start();
+
+    inited = true;
+}
+
+//*******************************************************************************
+void Analyzer::compute(int nframes, float** inputs, float** outputs)
+{
+    if (not inited) {
+        std::cerr << "*** Analyzer " << this << ": init never called! Doing it now.\n";
+        if (fSamplingFreq <= 0) {
+            fSamplingFreq = 48000;
+            std::cout << "Analyzer " << this
+                      << ": *** HAD TO GUESS the sampling rate (chose 48000 Hz) ***\n";
+        }
+        init(fSamplingFreq);
+    }
+
+    /* sum up all channels and add it to the buffer */
+    for (int i = 0; i < nframes; i++) {
+        float sum = 0;
+        for (int ch = 0; ch < mNumChannels; ch++) {
+            if (!mIsMonitoringAnalyzer) {
+                sum += inputs[ch][i];
+            } else {
+                sum += outputs[ch][i];
+            }
+        }
+        mCircularBuffer.push(sum);
+    }
+
+    hasProcessedAudio = true;
+}
+
+//*******************************************************************************
+void Analyzer::onTick()
+{
+    // cannot process audio if the no samples have been added to the ring buffer yet
+    if (!hasProcessedAudio) {
+        return;
+    }
+
+    const uint32_t samples = mCircularBuffer.size();
+    std::vector<float> fftBuffer;
```

Comment:
can we make this a class member so that resize() only performs memory operations when necessary to grow, versus every call to `onTick`?

Reply:
done